code/datums/martial/reverbpalm.dm

```diff
@@ -0,0 +1,286 @@
+	if(issilicon(L))
+		L.apply_damage(borgdam, BRUTE, limb_to_hit, armor, wound_bonus=CANT_WOUND)
+
+/datum/martial_art/reverberating_palm/proc/crash(atom/movable/ram, var/turf/Q)
+	if(Q.density || (!(Q.reachableTurftestdensity(T = Q))))
+		if(isliving(ram))
+			var/mob/living/target = ram
+			target.adjustBruteLoss(5)
+			if(isanimal(target))
+				if(target.stat == DEAD)
+					target.visible_message(span_warning("[target] crashes and explodes!"))
+					target.gib()
+		if(ismineralturf(Q))
+			var/turf/closed/mineral/M = Q
+			M.attempt_drill()
+	if(Q.density || (!(Q.reachableTurftestdensity(T = Q))))
+		return FALSE
+	else 
+		ram.forceMove(Q)
+		return TRUE
+
+//animation procs
+
+//knocking them down
+/datum/martial_art/reverberating_palm/proc/footsies(mob/living/target, var/suplex = FALSE)
+	if(suplex)
+		animate(target, transform = matrix(180, MATRIX_ROTATE), time = 0 SECONDS, loop = 0)
+	if(target.mobility_flags & MOBILITY_STAND)
+		animate(target, transform = matrix(90, MATRIX_ROTATE), time = 0 SECONDS, loop = 0)
+		return
+
+//Check for if someone is allowed to be stood back up
+/datum/martial_art/reverberating_palm/proc/wakeup(mob/living/target)
+	if(target.mobility_flags & MOBILITY_STAND)
+		animate(target, transform = null, time = 0.4 SECONDS, loop = 0)
+
+/datum/martial_art/reverberating_palm/proc/initiate(mob/living/user)
+	var/obj/effect/temp_visual/decoy/fading/onesecond/F = new(get_turf(user), user)
+	animate(F, alpha = 100, color = "#00d9ff")
+	walk_towards(F, user, 0, 1.5)
+	playsound(user,'sound/effects/gravhit.ogg', 20, 1)
+
+/datum/martial_art/reverberating_palm/can_use(mob/living/carbon/human/H)
+	var/obj/item/bodypart/r_arm/robot/seismic/R = H.get_bodypart(BODY_ZONE_R_ARM)
+	if(R)
+		if(!istype(R, /obj/item/bodypart/r_arm/robot/seismic))
+			return FALSE
+		if((R?.bodypart_disabled))
+			return FALSE
+	if(H.restrained() || H.get_active_held_item() || HAS_TRAIT(H, TRAIT_PACIFISM) || !(H.mobility_flags & MOBILITY_MOVE) || H.stat != CONSCIOUS)
+		return FALSE
+	return ..()
+
+/datum/martial_art/reverberating_palm/proc/InterceptClickOn(mob/living/carbon/human/H, params, atom/target)
+	var/list/modifiers = params2list(params)
+	if(!(can_use(H)) || (modifiers["shift"] || modifiers["alt"]))
+		return
+	H.face_atom(target)
+	if(H==target)
+		if(H.a_intent == INTENT_HELP)
+			supercharge(H)
+		return
+	if(H.a_intent == INTENT_DISARM)
+		rush(H)
+	if(H.a_intent == INTENT_HARM && isliving(target))
+		suplex(H,target)
+	if(H.a_intent == INTENT_GRAB)
+		lariat(H)
+
+/datum/martial_art/reverberating_palm/harm_act(mob/living/carbon/human/A, mob/living/D)
+	if(usr.click_intercept == null)
+		return FALSE // no punching plus slamming please
+
+/datum/martial_art/reverberating_palm/proc/supercharge(mob/living/user)
+	if(!COOLDOWN_FINISHED(src, next_palm))
+		to_chat(user, span_warning("You can't do that yet!"))
+		return
+	COOLDOWN_START(src, next_palm, COOLDOWN_RPALM)
```

Comment:
with the cooldown being here, it means if they accidentally click themselves, they'll waste the cooldown even if they cancel the charging, just want to make sure that's alright with you